source/mathPres/MathCAT/MathCATPreferences.py

```diff
@@ -0,0 +1,1016 @@
+	and navigation settings. Extends MathCATgui.MathCATPreferencesDialog.
+	"""
+
+	def __init__(self, parent: wx.Window | None):
+		"""Initialize the preferences dialog.
+
+		Sets up the UI, loads preferences, applies defaults and saved settings,
+		and restores the previous UI state.
+
+		:param parent: The parent window for the dialog.
+		"""
+		# initialize parent class
+		MathCATgui.MathCATPreferencesDialog.__init__(self, parent)
+
+		# load the logo into the dialog
+		fullPathToLogo: str = os.path.join(os.path.dirname(os.path.abspath(__file__)), "logo.png")
+		if os.path.exists(fullPathToLogo):
+			self._bitmapLogo.SetBitmap(wx.Bitmap(fullPathToLogo))
+
+		# load in the system values followed by the user prefs (if any)
+		UserInterface.loadDefaultPreferences()
+		UserInterface.loadUserPreferences()
+
+		# hack for "CopyAs" because its location in the prefs is not yet fixed
+		if "CopyAs" not in userPreferences["Navigation"]:
+			userPreferences["Navigation"]["CopyAs"] = (
+				userPreferences["Other"]["CopyAs"] if "CopyAs" in userPreferences["Other"] else "MathML"
+			)
+		UserInterface.validateUserPreferences()
+
+		if "NVDAAddOn" in userPreferences:
+			# set the categories selection to what we used on last run
+			self._listBoxPreferencesTopic.SetSelection(userPreferences["NVDAAddOn"]["LastCategory"])
+			# show the appropriate dialogue page
+			self._simplebookPanelsCategories.SetSelection(self._listBoxPreferencesTopic.GetSelection())
+		else:
+			# set the categories selection to the first item
+			self._listBoxPreferencesTopic.SetSelection(0)
+			userPreferences["NVDAAddOn"] = {"LastCategory": "0"}
+		# populate the languages and braille codes
+		UserInterface.getLanguages(self)
+		UserInterface.getBrailleCodes(self)
+		# set the ui items to match the preferences
+		UserInterface.setUIValues(self)
+
+	@staticmethod
+	def pathToLanguagesFolder() -> str:
+		r"""Returns the full path to the Languages rules folder.
+
+		The language rules are stored in:
+		MathCAT\Rules\Languages, relative to the location of this file.
+
+		:return: Absolute path to the Languages folder as a string.
+		"""
+		return os.path.join(os.path.dirname(os.path.abspath(__file__)), "Rules", "Languages")
+
+	@staticmethod
+	def pathToBrailleFolder() -> str:
+		r"""Returns the full path to the Braille rules folder.
+
+		The Braille rules are stored in:
+		MathCAT\Rules\Braille, relative to the location of this file.
+
+		:return: Absolute path to the Braille folder as a string.
+		"""
+		return os.path.join(os.path.dirname(os.path.abspath(__file__)), "Rules", "Braille")
+
+	@staticmethod
+	def languagesSet() -> frozenset[str]:
+		"""Returns a set of supported language codes.
+
+		This set includes standard language codes, as well as regional variants such as
+		'en-GB', 'zh-HANT', and others.
+
+		:return: A set frozenset of supported language codes (e.g., 'en', 'fr', 'zh-HANS').
+		"""
+		languages = frozenset(
+			[
+				"aa",
```

Comment:
There were descriptions added to these before. Can you please add them to `LANG_NAMES_TO_LOCALIZED_DESCS`. This is so we have fall back names for these language descriptions. You don't need to add them to `LANG_NAMES_TO_LOCALIZED_DESCS` if `getLanguageDescription` doesn't return None. You can test this from the NVDA python console by calling `getLanguageDescription` with each list item here. If any item returns None, add it to `LANG_NAMES_TO_LOCALIZED_DESCS` with the english name of the language.